core/arch/aarch64/instr_create.h

```diff
@@ -40,6 +40,26 @@
 
 /* DR_API EXPORT TOFILE dr_ir_macros_aarch64.h */
 /* DR_API EXPORT BEGIN */
+
+/**
+ * Used in an additional immediate source operand to a vector operation, denotes
+ * half-precision floating point vector elements. See \ref sec_IR_AArch64.
+ */
+#define FSZ_HALF 1
+
+/**
+ * Used in an additional immediate source operand to a vector operation, denotes
+ * single-precision floating point vector elements. See \ref sec_IR_AArch64.
+ */
+#define FSZ_SINGLE 2
+
+/**
+ * Used in an additional immediate source operand to a vector operation, denotes
+ * double-precision floating point vector elements. See \ref sec_IR_AArch64.
+ */
+#define FSZ_DOUBLE 3
```

Comment:
Question: are there INSTR_CREATE_ macros for vector operations or they haven't been added yet?  If they exist they should probably refer to \ref sec_IR_AArch64 as well.

Reply:
I plan to follow this up with a separate patch that adds INSTR_CREATE macros for the new instructions, together with a test program that uses it, if you are OK with that? 

As for the auto-generation, @AssadHashmi has a script that needs updating, but it will take some time until he has some spare time for that unfortunately, I think.

Reply:
Sure, seems reasonable.